src/Compiler/SyntaxTree/SyntaxTree.fsi

```diff
@@ -374,7 +374,12 @@ type SynBindingKind =
 
 /// Represents the explicit declaration of a type parameter
 [<NoEquality; NoComparison>]
-type SynTyparDecl = SynTyparDecl of attributes: SynAttributes * SynTypar
+type SynTyparDecl =
+    | SynTyparDecl of
+        attributes: SynAttributes *
+        typar: SynTypar *
+        intersectionConstraints: SynType list *
+        trivia: SynTyparDeclTrivia
```

Comment:
Intersection constraints are inseparable from explicit typar declarations on classes/members, so I think modelling it like this is better than trying to retrofit the constraints and their trivia on actual `SynTypeConstraint` in `SynTyparDecls`. The downside of this approach is the fact that we still need to create synthetic `SynTypeConstraint` during typechecking in order to reuse existing constraint checking mechanisms.

Adding `SynType.Intersection` vs reusing `SynType.WithGlobalConstraints` (for cases like `let f (x: 't & #IDisposable & seq<int>) = ()` where typars are not declared explicitly upfront) presents a similar dilemma.